$NetBSD: patch-cd,v 1.1 2007/04/01 23:50:59 hira Exp $

Security fixes for CVE-2007-0239.  This has been fixed in OOo 2.2.

--- shell/source/unix/misc/open-url.sh.orig	2006-12-02 02:57:03.000000000 +0900
+++ shell/source/unix/misc/open-url.sh	2007-01-19 22:07:49.000000000 +0900
@@ -96,15 +96,15 @@
 # special handling for mailto: uris
 if echo $1 | grep '^mailto:' > /dev/null; then
   # check $MAILER variable
-  if [ $MAILER ]; then
-    $MAILER $1 &
+  if [ ! -z "$MAILER" ]; then
+    $MAILER "$1" &
     exit 0
   else
     # mozilla derivates may need -remote semantics
     for i in thunderbird mozilla netscape; do
       mailer=`which $i`
       if [ ! -z "$mailer" ]; then
-        run_mozilla $mailer $1
+        run_mozilla "$mailer" "$1"
         exit 0
       fi
     done
@@ -113,15 +113,15 @@
   fi
 else
   # check $BROWSER variable
-  if [ $BROWSER ]; then
-    $BROWSER $1 &
+  if [ ! -z "$BROWSER" ]; then
+    $BROWSER "$1" &
     exit 0
   else
     # mozilla derivates may need -remote semantics
     for i in firefox mozilla netscape; do
       browser=`which $i`
       if [ ! -z "$browser" ]; then
-        run_mozilla $browser $1
+        run_mozilla "$browser" "$1"
         exit 0
       fi
     done
